make fake: record "new" has no field "charge_volume" CONTEXT:  PL/pgSQL function process_payday() line 3 at SQL statement

Hello. I would like to help with Gratipay development, but I got stuck at `make fake`:

```
$ make fake
env/bin/honcho run -e defaults.env,local.env env/bin/fake_data fake_data
Making Participants
Making Teams
Making Payment Instructions
Making Elsewheres
Making Communities
Making Tips
Making Transfers (5000/5000)
Making Paydays (1/53)Traceback (most recent call last):
  File "env/bin/fake_data", line 9, in <module>
    load_entry_point('gratipay==1913', 'console_scripts', 'fake_data')()
  File "/home/fenryxo/dev/projects/gratipay/gratipay.com/gratipay/utils/fake_data.py", line 387, in main
    populate_db(db)
  File "/home/fenryxo/dev/projects/gratipay/gratipay.com/gratipay/utils/fake_data.py", line 380, in populate_db
    _fake_thing(db, "paydays", **payday)
  File "/home/fenryxo/dev/projects/gratipay/gratipay.com/gratipay/utils/fake_data.py", line 34, in _fake_thing
    , column_values
  File "/home/fenryxo/dev/projects/gratipay/gratipay.com/env/lib/python2.7/site-packages/postgres/__init__.py", line 357, in run
    cursor.run(sql, parameters)
  File "/home/fenryxo/dev/projects/gratipay/gratipay.com/env/lib/python2.7/site-packages/postgres/cursors.py", line 92, in run
    self.execute(sql, parameters)
  File "/home/fenryxo/dev/projects/gratipay/gratipay.com/env/lib/python2.7/site-packages/psycopg2/extras.py", line 288, in execute
    return super(NamedTupleCursor, self).execute(query, vars)
psycopg2.ProgrammingError: record "new" has no field "charge_volume"
CONTEXT:  PL/pgSQL function process_payday() line 3 at SQL statement

Makefile:45: recipe for target 'fake' failed
make: *** [fake] Error 1
```

I'm running Debian Jessie with Python 2.7.9 and Postgresql 9.4.2, if that matters.

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/26649678-make-fake-record-new-has-no-field-charge_volume-context-pl-pgsql-function-process_payday-line-3-at-sql-statement?utm_campaign=plugin&utm_content=tracker%2F85909&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F85909&utm_medium=issues&utm_source=github).
</bountysource-plugin>

--- a/gratipay/utils/fake_data.py
+++ b/gratipay/utils/fake_data.py
@@ -381,13 +381,15 @@ def populate_db(db, num_participants=100, ntips=200, num_teams=5, num_transfers=
         date = end_date
     print("")
 
-def main():
-    db = wireup.db(wireup.env())
+
+def main(db, *a, **kw):
+    clean_db(db)
     prep_db(db)
-    populate_db(db)
+    populate_db(db, *a, **kw)
     clean_db(db)
     check_db(db)
 
 
 if __name__ == '__main__':
-    main()
+    db = wireup.db(wireup.env())
+    main(db)

--- a/gratipay/utils/fake_data.py
+++ b/gratipay/utils/fake_data.py
@@ -225,50 +225,12 @@ def prep_db(db):
 
         CREATE TRIGGER process_exchange AFTER INSERT ON exchanges
             FOR EACH ROW EXECUTE PROCEDURE process_exchange();
-
-        CREATE OR REPLACE FUNCTION process_payday() RETURNS trigger AS $$
-            BEGIN
-                SELECT COALESCE(SUM(amount+fee), 0)
-                  FROM exchanges
-                 WHERE timestamp > NEW.ts_start
-                   AND timestamp < NEW.ts_end
-                   AND amount > 0
-                  INTO NEW.charge_volume;
-
-                SELECT COALESCE(SUM(fee), 0)
-                  FROM exchanges
-                 WHERE timestamp > NEW.ts_start
-                   AND timestamp < NEW.ts_end
-                   AND amount > 0
-                  INTO NEW.charge_fees_volume;
-
-                SELECT COALESCE(SUM(amount), 0)
-                  FROM exchanges
-                 WHERE timestamp > NEW.ts_start
-                   AND timestamp < NEW.ts_end
-                   AND amount < 0
-                  INTO NEW.ach_volume;
-
-                SELECT COALESCE(SUM(fee), 0)
-                  FROM exchanges
-                 WHERE timestamp > NEW.ts_start
-                   AND timestamp < NEW.ts_end
-                   AND amount < 0
-                  INTO NEW.ach_fees_volume;
-
-                RETURN NEW;
-            END;
-        $$ language plpgsql;
-
-        CREATE TRIGGER process_payday BEFORE INSERT ON paydays
-            FOR EACH ROW EXECUTE PROCEDURE process_payday();
     """)
 
 def clean_db(db):
     db.run("""
-        DROP FUNCTION process_transfer() CASCADE;
-        DROP FUNCTION process_exchange() CASCADE;
-        DROP FUNCTION process_payday() CASCADE;
+        DROP FUNCTION IF EXISTS process_transfer() CASCADE;
+        DROP FUNCTION IF EXISTS process_exchange() CASCADE;
     """)
 
 

--- a/gratipay/utils/fake_data.py
+++ b/gratipay/utils/fake_data.py
@@ -344,7 +344,8 @@ def populate_db(db, num_participants=100, ntips=200, num_teams=5, num_transfers=
     print("")
 
 
-def main(db, *a, **kw):
+def main(db=None, *a, **kw):
+    db = db or wireup.db(wireup.env())
     clean_db(db)
     prep_db(db)
     populate_db(db, *a, **kw)
@@ -353,5 +354,4 @@ def main(db, *a, **kw):
 
 
 if __name__ == '__main__':
-    db = wireup.db(wireup.env())
-    main(db)
+    main()

--- a/tests/py/test_fake_data.py
+++ b/tests/py/test_fake_data.py
@@ -14,7 +14,7 @@ class TestFakeData(Harness):
         num_tips = 5
         num_teams = 1
         num_transfers = 5
-        fake_data.populate_db(self.db, num_participants, num_tips, num_teams, num_transfers)
+        fake_data.main(self.db, num_participants, num_tips, num_teams, num_transfers)
         tips = self.db.all("SELECT * FROM tips")
         participants = self.db.all("SELECT * FROM participants")
         transfers = self.db.all("SELECT * FROM transfers")

